modules/core/src/test/java/org/apache/ignite/internal/processors/cache/CacheGroupsMetricsRebalanceTest.java

```diff
@@ -192,25 +194,23 @@ public void testRebalance() throws Exception {
         final CountDownLatch l1 = new CountDownLatch(1);
         final CountDownLatch l2 = new CountDownLatch(1);
 
-        startGrid(4).events().localListen(new IgnitePredicate<Event>() {
-            @Override public boolean apply(Event evt) {
-                l1.countDown();
+        ignite = startGrid(1);
 
-                try {
-                    assertTrue(l2.await(5, TimeUnit.SECONDS));
-                }
-                catch (InterruptedException e) {
-                    throw new AssertionError();
-                }
+        ignite.events().localListen((IgnitePredicate<Event>)evt -> {
```

Comment:
This listener must be specified via `IgniteConfiguration` otherwise there is no guarantee that rebalancing is not already finished.